Anastasiya Santalova/hw4.js

```diff
@@ -0,0 +1,224 @@
+"use strict"
+
+/**
+* * * * * * * ORDER * * * * * * * 
+*/
+
+function Order () {
+    this.products = [];
+}
+
+/**
+* Accept data from added food
+*/
+
+Order.prototype.acceptHumburger = function (Hamburger) {
+    this.products.push(Hamburger);
+}
+
+Order.prototype.acceptSalad = function (Salad) {
+    this.products.push(Salad);
+}
+
+Order.prototype.acceptDrink = function (Drink) {
+    this.products.push(Drink);
+}
+
+/**
+* Delete data from order
+*/
+
+Order.prototype.deleteProduct = function (name) {
+    this.products = this.products.filter((arg) => arg !== name);
+}
+
+/**
+ * Get know a price
+ */
+Order.prototype.calculatePrice = function () {
+
+    this.totalPrice = this.products.reduce(function(acc, obj){
+        acc += obj.price;
+        return acc;
+    }, 0);
+
+}
+
+/**
+ * Get know calories
+ */
+
+Order.prototype.calculateCalories = function () {
+    
+    this.totalCalories = this.products.reduce(function(acc, obj){
+        acc += obj.calories;
+        return acc;
+    }, 0);
+
+}
+
+/**
+* Paid order
+*/
+
+Order.prototype.paid = function () {
+
+    this.products.forEach(Object.freeze);
+    Object.freeze(this.products);
+    Object.freeze(this);
+
+}
+
+
+/**
+* * * * * * * CLASS FOOD * * * * * * * 
+*/
+
+function Food(price, calories) {
+    this.price = price;
+    this.calories = calories;
+}
+
+
+/**
+* * * * * * * CLASS HAMBURGER * * * * * * * 
+*/
+
+function Hamburger(size, stuffing) {
+
+    this.size = size;
+    this.stuffing = stuffing;
+
+    var hamPrice, hamCalories, priceOfStuffing, caloriesOfStuffing;
+
+    if (stuffing[0][0]) {
```

Comment:
I'd rather at least combine two `if` statements together and maybe even moved all the duplicated calculations code into the separated function